src/stream.js

```diff
@@ -0,0 +1,488 @@
+      return null // no override needed
+    }
+
+    fetch = http.resolveFetch(null, false)
+    if (fetch === null) {
+      fetch = function() {
+        return Promise.reject(
+          new errors.StreamsNotSupported(
+            'Could not find a stream compatible fetch function. ' +
+              'Please, consider providing a Fetch API compatible function ' +
+              'with streamable response bodies.'
+          )
+        )
+      }
+    }
+
+    return fetch
+  }
+}
+
+/**
+ * Takes a snapshot of the current query. Assumes the subscribed query returns a
+ * reference.
+ */
+StreamClient.prototype.snapshot = function() {
+  var self = this
+  self._client
+    .query(q.Get(self._query))
+    .then(function(doc) {
+      self._onEvent({
+        event: 'snapshot',
+        data: doc,
+      })
+    })
+    .catch(function(error) {
+      self._onEvent({
+        event: 'error',
+        data: error,
+      })
+    })
+}
+
+/** Initiates the stream subscription.  */
+StreamClient.prototype.subscribe = function() {
+  var self = this
+  if (self._state === 'idle') {
+    self._state = 'open'
+  } else {
+    throw new Error('The stream alread started at once.')
+  }
+
+  var body = JSON.stringify(self._query)
+  var startTime = Date.now()
+
+  function handleResponse(response) {
+    var endTime = Date.now()
+    var parsed
+
+    if (response.ok) {
+      // There's no textual representation to be set here since streams are an
+      // unbounded event source, hence, the syntactic representation.
+      parsed = Promise.resolve({
+        text: '[stream]',
+        data: '[stream]',
+      })
+    } else {
+      parsed = response.text().then(function(text) {
+        return {
+          text: text,
+          data: json.parseJSON(text),
+        }
+      })
+    }
+
+    return parsed.then(function(parsed) {
+      var headers = http.responseHeadersAsObject(response)
+      var result = new RequestResult(
+        'POST',
+        'stream',
+        self._urlParams,
+        body,
+        self._query,
+        parsed.text,
+        parsed.data,
+        response.status,
+        headers,
+        startTime,
+        endTime
+      )
+      self._client._handleRequestResult(response, result)
+    })
+  }
+
+  function onData(data) {
+    var event = json.parseJSON(data)
+    if (event.txnTS !== undefined) {
+      self._client.syncLastTxnTime(event.txnTS)
+    }
+    if (event.event === 'error') {
+      onError(new errors.StreamErrorEvent(event))
+    } else {
+      self._onEvent(event)
+    }
+  }
+
+  function onError(error) {
+    // AbortError is triggered by the AbortController as result of calling
+    // close() on a Subscription. There's no need to relay this event back up.
+    if (error.name !== 'AbortError') {
+      self._onEvent({
+        event: 'error',
+        data: error,
+      })
+    }
+  }
+
+  // Minimum browser compatibility based on current code:
```

Comment:
@erickpintor Great documentation here 🚀

Reply:
have you tested all those browsers?

Reply:
Unfortunately, no. I don't have access to all those platforms. This information is derived by the functions called in the code and their minimum browser version documented for them at MDN web docs.